### PR TITLE
Updating to follow the tatest swift-otel

### PR DESCRIPTION
**Pull request by omochi**
As the title suggests, the following adjustments have been made:

- Updated to follow the type name changes in the `OTel` module.
- Modified the injection method for `RandomNumberGenerator` to match the style used in `swift-otel`.
- Updated to reflect the method name changes in the `OTelIDGenerator`P protocol.
- Copied and pasted `swift-otel`'s `Hex` for reuse. It appears that this was made `public` at one point, but it seems to have reverted back to `internal`.

### CODE REVIEW

File: Sources/OpenTelemetryXRay/IDGenerator.swift
Comment (slashmo) [inline Sources/OpenTelemetryXRay/IDGenerator.swift:27]: Given that this initializer doesn't have a default `RandomNumberGenerator` anymore, could we remove this bit of documentation, and instead document the public init you added in the extension below too?

File: Package.swift
Comment (slashmo) [inline Package.swift:11]: Nitpick here: Can we move the NIO dependency below the OTel dependency since OTel is the primary one?

**Review by slashmo — APPROVED** [review]
Thanks for creating the PR, looks great! 🙏

**Comment by slashmo** [comment]
The CI failures are my fault for not keeping up to date with the changes in the main repo. I'll create a PR soon to update the CI in this repo as well and ping you once it's merged so you can rebase your PR.

**Comment by omochi** [comment]
I have updated the points of note.

**Comment by slashmo** [comment]
Thanks for the reminder @omochi, this slipped my mind over the Christmas break. I'll tackle the CI issues today to unblock this PR, rebase it on top of the CI changes, and then merge it if everything's green. Thanks again for your work! 🙏

**Review by slashmo — APPROVED** [review]

**Comment by slashmo** [comment]
Thanks again for your contribution @omochi 🙏

**Comment by omochi** [comment]
Thank you!

**Pull request merged**
